### PR TITLE
ci: new workflow for testing examples

### PR DESCRIPTION
**Pull request by bhavsarpratik**
Changes needed after jina-ai/examples#252 is merged

### CODE REVIEW

**Review by hanxiao — CHANGES_REQUESTED** [review]
this change the current CD, please don't. Start a new workflow.

**Comment by bhavsarpratik** [comment]
Let me move `examples-test` job to a new workflow as it might require some updates/experiments and I am keeping its trigger on all `pull_request` to test it now and to see if any change breaks any example in the future. This will fail till jina-ai/examples#252 is merged

File: .github/workflows/test-examples.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: Test Examples
+
+on:
+  pull_request:
+
```
Comment (hanxiao) [inline .github/workflows/test-examples.yml:5]:
```suggestion
on:
  push:
    branches:
      - master

```

File: .github/workflows/test-examples.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
```
Comment (hanxiao) [inline .github/workflows/test-examples.yml:25]:
```suggestion      
      - uses: actions/checkout@v2
        with:
          repository: jina-ai/examples
          path: examples
      - name: Install dependencies
```

File: .github/workflows/test-examples.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        run: |
+          python -m pip install --upgrade pip
+          pip install .[cicd,test,http] requests --no-cache-dir
+      - name: Test example with pytest
+        run: |
+          jina check
+          export JINA_LOG_VERBOSITY="ERROR"
+          git clone https://github.com/jina-ai/examples.git
```
Comment (hanxiao) [inline .github/workflows/test-examples.yml:33]:
```suggestion
```

**Review by JoanFM — APPROVED** [review]

**Review by hanxiao — APPROVED** [review]

**Pull request merged**